docs/architecture/overview.md

```diff
@@ -47,8 +49,24 @@ The check before a transaction can e.g. programmatically check all of the parame
 > ⚠️ IMPORTANT: Since a guard has full power to block Safe transaction execution,
 > a broken guard can cause a denial of service for the Safe. Make sure to carefully audit the guard code and design recovery mechanisms.
 
+To learn more about guards, please refer to the [guards](./guards.md) documentation.
+
+#### Libraries
+
+Libraries implement custom execution flows that are optimized towards specific protocols or use cases. To use the libraries they are called using `delegatecall`, which executes the library logic via the Safe contract.
+
+> ⚠️ IMPORTANT: Since a libraries are called via delegate call, they can perform arbitrary state changes on the Safe contract. Make sure to carefully audit the libraries that are used.
```

Comment:
```suggestion
> ⚠️ IMPORTANT: Since libraries are called via delegate call, they can perform arbitrary state changes on the Safe contract. Make sure to carefully audit the libraries that are used.
```